src/Pim/Bundle/CatalogBundle/Doctrine/ORM/Repository/ProductUniqueDataRepository.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace Pim\Bundle\CatalogBundle\Doctrine\ORM\Repository;
+
+use Doctrine\ORM\EntityRepository;
+use Pim\Component\Catalog\Model\ProductInterface;
+use Pim\Component\Catalog\Model\ProductValueInterface;
+use Pim\Component\Catalog\Repository\ProductUniqueDataRepositoryInterface;
+
+/**
+ * Product unique data repository. Please see {@see Pim\Component\Catalog\Model\ProductUniqueDataInterface}
+ * for more information.
+ *
+ * @author    Julien Janvier <julien.janvier@akeneo.com>
+ * @copyright 2017 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class ProductUniqueDataRepository extends EntityRepository implements ProductUniqueDataRepositoryInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function uniqueDataNotExistsInThisProduct(ProductValueInterface $value, ProductInterface $product)
+    {
+        $queryBuilder = $this->createQueryBuilder('ud')
+            ->select('COUNT(ud)')
+            ->where('ud.attribute = :attribute')
+            ->andWhere('ud.rawData = :data')
+        ;
+
+        $parameters = [
+            'attribute' => $value->getAttribute(),
+            'data' => $value->__toString(),
+        ];
+
+        if (null !== $product->getId()) {
```

Comment:
@nidup 
as @grena suggested, I renamed the method to `uniqueDataExistsInAnotherProduct`
we do check for all products

which is more clear
what we want to do, is to check if the data already exists in another product
if the product has no id (can the case during a creation), we just skip the part where product.id != XXX